### PR TITLE
[SPARK-45907][CORE] Use  Java9+  ProcessHandle APIs to computeProcessTree in ProcfsMetricsGetter

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

This PR uses  Java9+ ProcessHandle APIs to computeProcessTree in ProcfsMetricsGetter.

### Why are the changes needed?

Simplify the code

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

new tests
### Was this patch authored or co-authored using generative AI tooling?

no

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/executor/ProcfsMetricsGetter.scala [hunk 1]
```diff
@@ -47,7 +46,7 @@ private[spark] class ProcfsMetricsGetter(procfsDir: String = "/proc/") extends L
   private val testing = Utils.isTesting
   private val pageSize = computePageSize()
   private var isAvailable: Boolean = isProcfsAvailable
-  private val pid = computePid()
+  private val pid = ProcessHandle.current()
```
Comment (LuciferYang) [inline core/src/main/scala/org/apache/spark/executor/ProcfsMetricsGetter.scala:49]: nit: Perhaps this variable should be renamed.

**Review by LuciferYang — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

It would be great if you can mention this is Java9+ API in the PR description .

**Comment by yaooqinn** [comment]
Thank you @dongjoon-hyun and @LuciferYang

**Comment by dongjoon-hyun** [comment]
Thank you for updatings. Merged to master~

**Pull request closed without merging**